compliance-by-design-demo/src/components/FactPrompt.js

```diff
@@ -48,13 +48,27 @@ class FactPrompt extends Component{
         return this.props.possibleCreatingActions.map(possibleCreatingAction => {
             const numberCandidates = this.props.previousActs.map(
                 (prevAct, index) => {
-                    return {index: index, ...prevAct} }
+                    return {index: index, ...prevAct, actorName: this._getActorName(prevAct)} }
                     ).filter(prevAct => prevAct.link === possibleCreatingAction)
-            const number = numberCandidates.length > 0 ? numberCandidates[0].index + 1 : "Unknown act"
-            return <option value={possibleCreatingAction}>{number}</option>
+            const number = numberCandidates.length > 0 ? `Act ${numberCandidates[0].index + 1}` : "Unknown act"
+            const name = numberCandidates.length > 0 ? numberCandidates[0].actorName : "Unknown actor"
+            return <option value={possibleCreatingAction}>{number} from {name}</option>
         })
     }
 
+    _getActorDid(act) {
+        const expression = Object.values(act.facts).find((fact) => fact.expression === "IS")
```

Comment:
This seems a little bit hacky. In practice, we do currently identify actors with IS, but in theory it's fine to not specify the actor facts in this way. 

What I think makes sense is using `getDidOfLinkedClaim` in core, maybe even doing this in law-reg rather than here, but I'll leave that to you.